core/blockchain.go

```diff
@@ -1681,16 +1664,9 @@ func (bc *BlockChain) insertChain(chain types.Blocks, setHead bool, makeWitness
 	switch {
 	// First block is pruned
 	case errors.Is(err, consensus.ErrPrunedAncestor):
-		if setHead {
-			// First block is pruned, insert as sidechain and reorg only if TD grows enough
-			log.Debug("Pruned ancestor, inserting as sidechain", "number", block.Number(), "hash", block.Hash())
-			return bc.insertSideChain(block, it, makeWitness)
```

Comment:
Insertion is now driven by the downloader/skeleton chain, which should always be inserting batches of blocks starting right above the chain head, and so it seems impossible that the parent state could be pruned here.

Reply:
But the `import` command doesn't provide this guarantee.... so the change here is incorrect.